pkg/source/apache-airflow/dag-processor-manager/parser.go

```diff
@@ -0,0 +1,207 @@
+	dagFilePath   string
+	logTask       taskid.TaskReference[[]*log.LogEntity]
+	targetLogType enum.LogType
+}
+
+func NewAirflowDagProcessorParser(dagFilePath string, logTask taskid.TaskReference[[]*log.LogEntity], targetLogType enum.LogType) *AirflowDagProcessorParser {
+	return &AirflowDagProcessorParser{
+		dagFilePath:   dagFilePath,
+		logTask:       logTask,
+		targetLogType: targetLogType,
+	}
+}
+
+// TargetLogType implements parser.Parser.
+func (a *AirflowDagProcessorParser) TargetLogType() enum.LogType {
+	return a.targetLogType
+}
+
+var _ parser.Parser = (*AirflowDagProcessorParser)(nil)
+
+func (*AirflowDagProcessorParser) Dependencies() []taskid.UntypedTaskReference {
+	return []taskid.UntypedTaskReference{}
+}
+
+func (*AirflowDagProcessorParser) Description() string {
+	return "The DagProcessorManager logs contain information for investigating the number of DAGs included in each Python file and the time it took to parse them. You can get information about missing DAGs and load."
+}
+
+func (*AirflowDagProcessorParser) GetParserName() string {
+	return "Airflow DagProcessorManager"
+}
+
+// Grouper implements parser.Parser.
+func (*AirflowDagProcessorParser) Grouper() grouper.LogGrouper {
+	return grouper.AllDependentLogGrouper
+}
+
+func (a *AirflowDagProcessorParser) LogTask() taskid.TaskReference[[]*log.LogEntity] {
+	return a.logTask
+}
+
+func (a *AirflowDagProcessorParser) Parse(ctx context.Context, l *log.LogEntity, cs *history.ChangeSet, builder *history.Builder) error {
+	textPayload, _ := l.MainMessage()
+
+	dagFileProcessorStats := a.fromLogEntity(textPayload)
+	if dagFileProcessorStats == nil {
+		// this is not a dag file processor stats log, skip
+		return nil
+	}
+	cs.RecordRevision(resourcepath.DagFileProcessorStats(dagFileProcessorStats), &history.StagingResourceRevision{
+		Verb:       enum.RevisionVerbComposerTaskInstanceStats,
+		State:      enum.RevisionStateConditionTrue,
+		Requestor:  "dag-processor-manager",
+		ChangeTime: l.Timestamp(),
+		Partial:    false,
+		Body: fmt.Sprintf("dags: %s\nerrors: %s",
+			dagFileProcessorStats.NumberOfDags(), dagFileProcessorStats.NumberOfErrors()),
+	})
+
+	// Emphasize "Error" for parsing dag failures
+	if dagFileProcessorStats.NumberOfErrors() != "0" {
+		cs.RecordLogSeverity(enum.SeverityError)
+	}
+
+	var summary string
+	if dagFileProcessorStats.Runtime() != "" {
+		summary = fmt.Sprintf("dags=%s, errors=%s, runtime=%s",
+			dagFileProcessorStats.NumberOfDags(), dagFileProcessorStats.NumberOfErrors(), dagFileProcessorStats.Runtime())
+	} else {
+		summary = fmt.Sprintf("dags=%s, errors= %s",
+			dagFileProcessorStats.NumberOfDags(), dagFileProcessorStats.NumberOfErrors())
+	}
+	cs.RecordLogSummary(summary)
+	return nil
+}
+
+// parse DAG Processor Manager's parse result log.
+// Sample: /home/airflow/gcs/dags/main.py 40441 4.06s 64 0 6.93s 2024-05-02T05:14:54
+func (a *AirflowDagProcessorParser) fromLogEntity(log string) *model.DagFileProcessorStats {
+
```

Comment:
I believe this should be refactored with using more generalized table parser with reading the past log lines.

But it would take some amount of time. We can do it later.

Reply:
you right, I added our offline discussion as a TODO comment.